File: 프로그래머스/2단계/구명 보트.js
function solution(people, limit) {
  people.sort((a, b) => b - a);
  let max = 0;
  let min = people.length - 1;
  let cnt = 0;

  while (max < min) {
    if (people[max] + people[min] <= limit) {
      max++;
      min--;
    } else max++;
    cnt++;
  }

  if (max == min) cnt++;
  return cnt;
}

console.log(solution([70, 50, 80, 50], 100));

File: 프로그래머스/lv2/42885. 구명보트/구명보트.py
def solution(people, limit):
    cnt = 0
    people.sort()
    
    p_min = 0
    p_max = len(people) - 1
    
    while p_min <= p_max:
        if p_min == p_max:
            cnt += 1
            break
            
        if people[p_min] + people[p_max] <= limit:
            p_min += 1
            p_max -= 1
        
        else:
            p_max -= 1
        cnt += 1
    return cnt